### DISCUSSION TITLE
关于一种可能的优化 Bilibili 网页卡顿的方法

### DISCUSSION THREAD

**Discussion by Tinhone** [the1812/Bilibili-Evolved#4147] [综合讨论 / General]
![图片](https://user-images.githubusercontent.com/81268244/232874357-d6c67f92-636f-487f-9ece-5fd0b0098f31.png)
https://www.bilibili.com/video/BV1Ev4y1J7qi/#reply160493842816
如图，以脚本的权限，是否有可能新增一个针对这方面做优化的插件🤔

**Comment by luotianyiismywife** [comment]
谢谢，试了一下成功了，内存占用成功下来了

**Reply by Tinhone** [reply]
😮

**Comment by the1812** [comment]
脚本拦不了, 得靠 uBlock 或者类似的插件

**Reply by Tinhone** [reply]
😯

**Reply by WakelessSloth56** [reply]
可以做到，直接使用 `Object.defineProperty` 重新给 `unsafeWindow` 定义 `Sentry` 属性，需要保留 `init` 和 `Integrations` 的 `Vue`，并且 `writable` 设置为 `false`。

```javascript
Object.defineProperty(unsafeWindow, 'Sentry', {
    value: {
        init: () => {},
        Integrations: {
            Vue: function () {},
        },
    },
    writable: false,
});
```

（没有深入测试，简单试了试除了控制台大片红之外，不影响什么功能。）

**Reply by Tinhone** [reply]
> 可以做到，直接使用 `Object.defineProperty` 重新给 `unsafeWindow` 定义 `Sentry` 属性，需要保留 `init` 和 `Integrations` 的 `Vue`，并且 `writable` 设置为 `false`。
> 
> ```js
> Object.defineProperty(unsafeWindow, 'Sentry', {
>     value: {
>         init: () => {},
>         Integrations: {
>             Vue: function () {},
>         },
>     },
>     writable: false,
> });
> ```
> 
> （没有深入测试，简单试了试除了控制台大片红之外，不影响什么功能。）

把 unsafeWindow 改成 window 粘贴进控制台运行试了一下，发现新版播放页翻评论的时候确实丝滑了一些，响应速度也快了一点。平均帧数要快 2 帧（运行前 8fps，运行后 10fps），最低帧数从 3 提升到了 7。内存占用相对来说差不多，都是1GB左右。控制台没有输出错误

**Reply by Tinhone** [reply]
> > 可以做到，直接使用 `Object.defineProperty` 重新给 `unsafeWindow` 定义 `Sentry` 属性，需要保留 `init` 和 `Integrations` 的 `Vue`，并且 `writable` 设置为 `false`。
> > ```js
> > Object.defineProperty(unsafeWindow, 'Sentry', {
> >     value: {
> >         init: () => {},
> >         Integrations: {
> >             Vue: function () {},
> >         },
> >     },
> >     writable: false,
> > });
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > （没有深入测试，简单试了试除了控制台大片红之外，不影响什么功能。）
> 
> 把 unsafeWindow 改成 window 粘贴进控制台运行试了一下，发现新版播放页翻评论的时候确实丝滑了一些，响应速度也快了一点。平均帧数要快 2 帧（运行前 8fps，运行后 10fps），最低帧数从 3 提升到了 7。内存占用相对来说差不多，都是1GB左右。控制台没有输出错误

旧版播放页的内存占用大约在 300MB~500MB 上下

**Comment by u2shana** [comment]
实测拦截之后搜索历史记录,还有搜索联想会失效
![image](https://user-images.githubusercontent.com/14146754/234212503-e725f16e-6f4f-4e8d-bac4-fe01bc7d8605.png)

**Reply by Tinhone** [reply]
😰

**Reply by luotianyiismywife** [reply]
我这边是没有影响的

**Reply by YaeSakuraQvQ** [reply]
搜索联想会变为`加载中...`

**Comment by Tinhone** [comment]
参照 #4201，看起来 B 站的性能问题越来越严重了...
同时，我这边浏览直播间页面，CPU 占用总是飙到 50% 以上
☹

**Reply by u2shana** [reply]
可能是因为你软解了hevc，现在B站直播间好像无论带不带pro的画质选项都是hevc，解决方法：
1：系统安装hevc插件，现在新版chrome和edge都支持原生调用hevc硬解了
2：找油猴脚本强制切换avc

**Reply by Tinhone** [reply]
> 可能是因为你软解了hevc，现在B站直播间好像无论带不带pro的画质选项都是hevc，解决方法： 1：系统安装hevc插件，现在新版chrome和edge都支持原生调用hevc硬解了 2：找油猴脚本强制切换avc

遗憾的是，安装了[这个脚本](https://greasyfork.org/zh-CN/scripts/459655)，已经确认直播间使用 avc 解码，但 CPU 占用依旧居高不下 :(

**Reply by Tinhone** [reply]
使用 Firefox 性能工具进行性能记录：
![图片](https://github.com/the1812/Bilibili-Evolved/assets/81268244/dc1780d8-8c1b-42c3-abfa-aeb2d85c4403)
![图片](https://github.com/the1812/Bilibili-Evolved/assets/81268244/246d8841-cfea-4457-b1e6-3d3622d6779f)
